lib/letters.ml

```diff
@@ -85,59 +87,63 @@ let certs_of_pem_directory ?(ext = "crt") path =
+  try
+    let open Mrmime in
+    let subject = Unstructured.Craft.v subject in
+    let date = Date.of_ptime ~zone:Date.Zone.GMT (Ptime_clock.now ()) in
+    let from_addr =
+      match Mailbox.of_string from with
+      | Ok v -> v
+      | Error _ -> raise (Invalid_email_address from)
     in
-    Header.of_list
-      Field.
-        [
-          Field (Field_name.content_type, Content, content1);
-          Field (Field_name.content_encoding, Encoding, `Quoted_printable);
-        ]
-  in
-  let html_headers =
-    let content1 =
-      let open Content_type in
-      make `Text
-        (Subtype.v `Text "html")
-        Parameters.(of_list [ (k "charset", v "utf-8") ])
+    let to_addresses = List.filter_map to_recipient_to_address recipients in
+    let cc_addresses = List.filter_map cc_recipient_to_address recipients in
+    let headers =
+      [
+        Field.(Field (Field_name.subject, Unstructured, subject));
+        Field.(Field (Field_name.date, Date, date));
+        Field.(Field (Field_name.from, Mailbox, from_addr));
+        Field.(Field (Field_name.v "To", Addresses, to_addresses));
+        Field.(Field (Field_name.cc, Addresses, cc_addresses));
+      ]
     in
-    Header.of_list
-      Field.
-        [
-          Field (Field_name.content_type, Content, content1);
-          Field (Field_name.content_encoding, Encoding, `Quoted_printable);
-        ]
-  in
-  let body =
-    match body with
-    | Plain text -> Mt.part ~header:plain_text_headers (stream_of_string text)
-    | Html text -> Mt.part ~header:html_headers (stream_of_string text)
-  in
-  Mt.make (Mrmime.Header.of_list headers) Mt.simple body
+    let plain_text_headers =
+      let content1 =
+        let open Content_type in
+        make `Text
+          (Subtype.v `Text "plain")
+          Parameters.(of_list [ (k "charset", v "utf-8") ])
+      in
+      Header.of_list
+        Field.
+          [
+            Field (Field_name.content_type, Content, content1);
+            Field (Field_name.content_encoding, Encoding, `Quoted_printable);
+          ]
+    in
+    let html_headers =
+      let content1 =
+        let open Content_type in
+        make `Text
+          (Subtype.v `Text "html")
+          Parameters.(of_list [ (k "charset", v "utf-8") ])
+      in
+      Header.of_list
+        Field.
+          [
+            Field (Field_name.content_type, Content, content1);
+            Field (Field_name.content_encoding, Encoding, `Quoted_printable);
+          ]
+    in
+    let body =
+      match body with
+      | Plain text -> Mt.part ~header:plain_text_headers (stream_of_string text)
+      | Html text -> Mt.part ~header:html_headers (stream_of_string text)
+    in
+    Ok (Mt.make (Mrmime.Header.of_list headers) Mt.simple body)
+  with
+  | Invalid_email_address address -> Error (Printf.sprintf "Invalid email address: %s" address)
+  | _ -> Error "Unexpected error while trying to build an email message"
```

Comment:
Use `Printexc.to_string` to stringify the exception, so we know what we went wrong.

Reply:
Fixed and thanks, I was wondering how to do this.

Reply:
I don't see the change in github

Reply:
damn, I somehow managed to fail to push it. Fixed, but now Github has some issues and cannot push.

Reply:
and now it worked, so fixed.